// @decodo/sdk-ts/build/esm/generated/parameters.js
// Auto-generated by src/codegen/web-scraping-api/generate-parameters.ts — do not edit
export const parameterMeta = {
    "360": { type: "boolean" },
    callback_url: { type: "string" },
    query: { type: "string", maxLength: 2048 },
    headless: { type: "string", enum: ["html", "png"] },
    locale: { type: "string" },
    geo: { type: "string" },
    device_type: { type: "string" },
    page_from: { type: "number", minimum: 0, maximum: 100 },
    google_results_language: { type: "string" },
    google_tbm: { type: "string" },
    google_tbs: { type: "string" },
    parse: { type: "boolean" },
    google_nfpr: { type: "boolean" },
    google_safe_search: { type: "boolean" },
    session_id: { type: "string" },
    xhr: { type: "boolean" },
    markdown: { type: "boolean" },
    page_count: { type: "number", minimum: 1, maximum: 10 },
    date_range: { type: "string" },
    stars: { type: "number" },
    adults: { type: "number", minimum: 1, maximum: 3 },
    children: { type: "number", minimum: 1, maximum: 3 },
    search_type: { type: "string" },
    date_start: { type: "string" },
    date_end: { type: "string" },
    url: { type: "string", maxLength: 2056 },
    hotel_occupancy: { type: "string" },
    domain: { type: "string" },
    language_code: { type: "string" },
    transcript_origin: { type: "string" },
    autoselect_variant: { type: "boolean" },
    currency: { type: "string" },
    category: { type: "string" },
    merchant: { type: "string" },
    sort_by: { type: "string" },
    parser_type: { type: "string" },
    product_id: { type: "string" },
    fulfillment_type: { type: "string" },
    walmart_store_id: { type: "string" },
    delivery_zip: { type: "string" },
    store_id: { type: "string" },
    delivery_type: { type: "string" },
    target_store_id: { type: "string" },
    lowes_store_id: { type: "string" },
    user_agent_type: { type: "string" },
    free_delivery: { type: "boolean" },
    pickup_today: { type: "boolean" },
    delivery_today_tomorrow: { type: "boolean" },
    payload: { type: "string" },
    proxy_pool: { type: "string", enum: ["standard", "premium"] },
    http_method: { type: "string" },
    successful_status_codes: { type: "array", items: { type: "number" } },
    headers: { type: "object" },
    cookies: { type: "object" },
    force_headers: { type: "boolean" },
    force_cookies: { type: "boolean" },
    prompt: { type: "string", maxLength: 8192 },
    search: { type: "boolean" },
    country: { type: "string" },
    sort: { type: "string" },
    upload_date: { type: "string" },
    type: { type: "string" },
    duration: { type: "string" },
    video_sort_by: { type: "string" },
    "3d": { type: "boolean" },
    "4k": { type: "boolean" },
    creative_commons: { type: "boolean" },
    hd: { type: "boolean" },
    hdr: { type: "boolean" },
    vr180: { type: "boolean" },
    live: { type: "boolean" },
    location: { type: "boolean" },
    purchased: { type: "boolean" },
    subtitles: { type: "boolean" },
    subtitle_origin: { type: "string" },
    limit: { type: "number", minimum: 0, maximum: 100 },
};
