// @decodo/sdk-ts/build/esm/generated/targets.js
// Auto-generated by src/codegen/web-scraping-api/generate-targets.ts — do not edit
/**
 * Scrape target discriminator (wire string values).
 */
export var Target;
(function (Target) {
    Target["UniversalEcommerce"] = "universal_ecommerce";
    Target["GoogleSearch"] = "google_search";
    Target["GoogleTravelHotels"] = "google_travel_hotels";
    Target["GoogleTrendsExplore"] = "google_trends_explore";
    Target["GoogleShoppingSearch"] = "google_shopping_search";
    Target["GoogleShoppingProduct"] = "google_shopping_product";
    Target["Google"] = "google";
    Target["GoogleSuggest"] = "google_suggest";
    Target["GoogleMaps"] = "google_maps";
    Target["GoogleAiMode"] = "google_ai_mode";
    Target["GoogleAds"] = "google_ads";
    Target["GoogleLens"] = "google_lens";
    Target["BingSearch"] = "bing_search";
    Target["Bing"] = "bing";
    Target["YoutubeTranscript"] = "youtube_transcript";
    Target["AmazonProduct"] = "amazon_product";
    Target["AmazonPricing"] = "amazon_pricing";
    Target["AmazonSearch"] = "amazon_search";
    Target["AmazonSellers"] = "amazon_sellers";
    Target["AmazonBestsellers"] = "amazon_bestsellers";
    Target["Amazon"] = "amazon";
    Target["Ecommerce"] = "ecommerce";
    Target["WalmartProduct"] = "walmart_product";
    Target["WalmartSearch"] = "walmart_search";
    Target["Walmart"] = "walmart";
    Target["TargetProduct"] = "target_product";
    Target["TargetSearch"] = "target_search";
    Target["Target"] = "target";
    Target["LowesSearch"] = "lowes_search";
    Target["Universal"] = "universal";
    Target["Chatgpt"] = "chatgpt";
    Target["Perplexity"] = "perplexity";
    Target["Gemini"] = "gemini";
    Target["Bbb"] = "bbb";
    Target["Autotrader"] = "autotrader";
    Target["Mobile"] = "mobile";
    Target["Airbnb"] = "airbnb";
    Target["AppleAppStore"] = "apple_app_store";
    Target["InstagramGraphqlProfile"] = "instagram_graphql_profile";
    Target["TiktokPost"] = "tiktok_post";
    Target["TiktokShopSearch"] = "tiktok_shop_search";
    Target["TiktokShopProduct"] = "tiktok_shop_product";
    Target["Tiktok"] = "tiktok";
    Target["RedditPost"] = "reddit_post";
    Target["RedditSubreddit"] = "reddit_subreddit";
    Target["RedditUser"] = "reddit_user";
    Target["YoutubeVideo"] = "youtube_video";
    Target["YoutubeMetadata"] = "youtube_metadata";
    Target["YoutubeSearch"] = "youtube_search";
    Target["YoutubeSearchMax"] = "youtube_search_max";
    Target["YoutubeSubtitles"] = "youtube_subtitles";
    Target["YoutubeChannel"] = "youtube_channel";
})(Target || (Target = {}));
export const targets = [Target.UniversalEcommerce, Target.GoogleSearch, Target.GoogleTravelHotels, Target.GoogleTrendsExplore, Target.GoogleShoppingSearch, Target.GoogleShoppingProduct, Target.Google, Target.GoogleSuggest, Target.GoogleMaps, Target.GoogleAiMode, Target.GoogleAds, Target.GoogleLens, Target.BingSearch, Target.Bing, Target.YoutubeTranscript, Target.AmazonProduct, Target.AmazonPricing, Target.AmazonSearch, Target.AmazonSellers, Target.AmazonBestsellers, Target.Amazon, Target.Ecommerce, Target.WalmartProduct, Target.WalmartSearch, Target.Walmart, Target.TargetProduct, Target.TargetSearch, Target.Target, Target.LowesSearch, Target.Universal, Target.Chatgpt, Target.Perplexity, Target.Gemini, Target.Bbb, Target.Autotrader, Target.Mobile, Target.Airbnb, Target.AppleAppStore, Target.InstagramGraphqlProfile, Target.TiktokPost, Target.TiktokShopSearch, Target.TiktokShopProduct, Target.Tiktok, Target.RedditPost, Target.RedditSubreddit, Target.RedditUser, Target.YoutubeVideo, Target.YoutubeMetadata, Target.YoutubeSearch, Target.YoutubeSearchMax, Target.YoutubeSubtitles, Target.YoutubeChannel];
export const targetMeta = {
    [Target.UniversalEcommerce]: {
        group: "None",
        response_format: "html",
        parameters: ["callback_url"],
    },
    [Target.GoogleSearch]: {
        group: "Google",
        response_format: "json",
        parameters: ["query", "headless", "locale", "geo", "device_type", "page_from", "google_results_language", "google_tbm", "google_tbs", "parse", "google_nfpr", "google_safe_search", "session_id", "xhr", "markdown", "page_count", "callback_url"],
    },
    [Target.GoogleTravelHotels]: {
        group: "Google",
        response_format: "html",
        parameters: ["query", "headless", "locale", "device_type", "page_from", "date_range", "stars", "adults", "children", "session_id", "markdown", "callback_url"],
    },
    [Target.GoogleTrendsExplore]: {
        group: "Google",
        response_format: "json",
        parameters: ["query", "geo", "device_type", "search_type", "date_start", "date_end", "callback_url"],
    },
    [Target.GoogleShoppingSearch]: {
        group: "Google",
        response_format: "json",
        parameters: ["query", "headless", "locale", "geo", "device_type", "page_from", "google_tbs", "parse", "session_id", "google_results_language", "markdown", "callback_url"],
    },
    [Target.GoogleShoppingProduct]: {
        group: "Google",
        response_format: "json",
        parameters: ["query", "headless", "locale", "geo", "device_type", "page_from", "parse", "session_id", "google_results_language", "markdown", "xhr", "callback_url"],
    },
    [Target.Google]: {
        group: "Google",
        response_format: "json",
        parameters: ["url", "headless", "locale", "device_type", "parse", "session_id", "markdown", "xhr", "page_count", "callback_url"],
    },
    [Target.GoogleSuggest]: {
        group: "Google",
        response_format: "json",
        parameters: ["query", "device_type", "geo", "locale", "session_id", "callback_url"],
    },
    [Target.GoogleMaps]: {
        group: "Google",
        response_format: "html",
        parameters: ["query", "headless", "geo", "locale", "page_from", "device_type", "session_id", "google_results_language", "google_nfpr", "hotel_occupancy", "date_range", "markdown", "callback_url"],
    },
    [Target.GoogleAiMode]: {
        group: "AI Tools",
        response_format: "json",
        parameters: ["query", "geo", "parse", "device_type", "session_id", "markdown", "xhr", "callback_url"],
    },
    [Target.GoogleAds]: {
        group: "Google",
        response_format: "json",
        parameters: ["query", "headless", "locale", "geo", "device_type", "page_from", "google_results_language", "google_tbm", "google_tbs", "parse", "google_nfpr", "session_id", "markdown", "xhr", "page_count", "callback_url"],
    },
    [Target.GoogleLens]: {
        group: "Google",
        response_format: "json",
        parameters: ["query", "headless", "parse", "device_type", "markdown", "callback_url"],
    },
    [Target.BingSearch]: {
        group: "Bing",
        response_format: "json",
        parameters: ["query", "headless", "locale", "geo", "domain", "device_type", "page_from", "parse", "page_count", "session_id", "markdown", "xhr", "callback_url"],
    },
    [Target.Bing]: {
        group: "Bing",
        response_format: "json",
        parameters: ["url", "headless", "locale", "geo", "device_type", "page_from", "parse", "session_id", "markdown", "xhr", "callback_url"],
    },
    [Target.YoutubeTranscript]: {
        group: "YouTube",
        response_format: "json",
        parameters: ["query", "language_code", "transcript_origin", "callback_url"],
    },
    [Target.AmazonProduct]: {
        group: "Amazon",
        response_format: "json",
        parameters: ["query", "headless", "domain", "device_type", "parse", "autoselect_variant", "geo", "session_id", "currency", "markdown", "xhr", "callback_url"],
    },
    [Target.AmazonPricing]: {
        group: "Amazon",
        response_format: "json",
        parameters: ["query", "headless", "domain", "device_type", "page_from", "parse", "geo", "session_id", "currency", "markdown", "xhr", "callback_url"],
    },
    [Target.AmazonSearch]: {
        group: "Amazon",
        response_format: "json",
        parameters: ["query", "headless", "domain", "device_type", "page_from", "category", "merchant", "parse", "geo", "session_id", "sort_by", "currency", "markdown", "xhr", "callback_url"],
    },
    [Target.AmazonSellers]: {
        group: "Amazon",
        response_format: "json",
        parameters: ["query", "headless", "locale", "domain", "device_type", "geo", "parse", "markdown", "xhr", "callback_url"],
    },
    [Target.AmazonBestsellers]: {
        group: "Amazon",
        response_format: "json",
        parameters: ["query", "domain", "device_type", "geo", "page_from", "category", "parse", "session_id", "currency", "markdown", "xhr", "callback_url"],
    },
    [Target.Amazon]: {
        group: "Amazon",
        response_format: "json",
        parameters: ["url", "headless", "device_type", "parse", "geo", "session_id", "markdown", "xhr", "callback_url"],
    },
    [Target.Ecommerce]: {
        group: "Other eCommerce",
        response_format: "json",
        parameters: ["url", "headless", "locale", "geo", "device_type", "parse", "parser_type", "callback_url"],
    },
    [Target.WalmartProduct]: {
        group: "Walmart",
        response_format: "html",
        parameters: ["product_id", "headless", "parse", "xhr", "markdown", "fulfillment_type", "walmart_store_id", "delivery_zip", "callback_url"],
    },
    [Target.WalmartSearch]: {
        group: "Walmart",
        response_format: "json",
        parameters: ["query", "headless", "parse", "markdown", "fulfillment_type", "walmart_store_id", "delivery_zip", "callback_url"],
    },
    [Target.Walmart]: {
        group: "Walmart",
        response_format: "html",
        parameters: ["url", "headless", "locale", "geo", "device_type", "store_id", "markdown", "xhr", "callback_url"],
    },
    [Target.TargetProduct]: {
        group: "Target",
        response_format: "json",
        parameters: ["product_id", "headless", "parse", "device_type", "markdown", "xhr", "delivery_type", "target_store_id", "delivery_zip", "callback_url"],
    },
    [Target.TargetSearch]: {
        group: "Target",
        response_format: "json",
        parameters: ["query", "headless", "parse", "device_type", "delivery_type", "target_store_id", "delivery_zip", "xhr", "markdown", "callback_url"],
    },
    [Target.Target]: {
        group: "Target",
        response_format: "html",
        parameters: ["url", "headless", "device_type", "xhr", "delivery_zip", "target_store_id", "callback_url"],
    },
    [Target.LowesSearch]: {
        group: "Lowe's",
        response_format: "json",
        parameters: ["query", "lowes_store_id", "headless", "delivery_zip", "user_agent_type", "free_delivery", "pickup_today", "delivery_today_tomorrow", "callback_url"],
    },
    [Target.Universal]: {
        group: "Universal",
        response_format: "html",
        parameters: ["url", "payload", "proxy_pool", "http_method", "headless", "geo", "locale", "device_type", "session_id", "successful_status_codes", "headers", "cookies", "force_headers", "force_cookies", "xhr", "markdown", "callback_url"],
    },
    [Target.Chatgpt]: {
        group: "AI Tools",
        response_format: "json",
        parameters: ["prompt", "search", "parse", "geo", "device_type", "markdown", "xhr", "callback_url"],
    },
    [Target.Perplexity]: {
        group: "AI Tools",
        response_format: "json",
        parameters: ["prompt", "parse", "geo", "device_type", "markdown", "xhr", "callback_url"],
    },
    [Target.Gemini]: {
        group: "AI Tools",
        response_format: "json",
        parameters: ["prompt", "parse", "geo", "xhr", "callback_url"],
    },
    [Target.Bbb]: {
        group: "Business Reviews",
        response_format: "html",
        parameters: ["url", "headless", "geo", "device_type", "markdown", "xhr", "callback_url"],
    },
    [Target.Autotrader]: {
        group: "Marketplace",
        response_format: "html",
        parameters: ["url", "headless", "geo", "device_type", "markdown", "xhr", "callback_url"],
    },
    [Target.Mobile]: {
        group: "Marketplace",
        response_format: "html",
        parameters: ["url", "headless", "geo", "device_type", "markdown", "xhr", "callback_url"],
    },
    [Target.Airbnb]: {
        group: "Travel",
        response_format: "html",
        parameters: ["url", "headless", "geo", "device_type", "markdown", "xhr", "callback_url"],
    },
    [Target.AppleAppStore]: {
        group: "Marketplace",
        response_format: "html",
        parameters: ["url", "headless", "geo", "device_type", "markdown", "xhr", "callback_url"],
    },
    [Target.InstagramGraphqlProfile]: {
        group: "Instagram",
        response_format: "json",
        parameters: ["query", "callback_url"],
    },
    [Target.TiktokPost]: {
        group: "TikTok",
        response_format: "html",
        parameters: ["url", "xhr", "callback_url"],
    },
    [Target.TiktokShopSearch]: {
        group: "TikTok",
        response_format: "html",
        parameters: ["query", "headless", "device_type", "markdown", "country", "callback_url"],
    },
    [Target.TiktokShopProduct]: {
        group: "TikTok",
        response_format: "html",
        parameters: ["product_id", "headless", "device_type", "xhr", "markdown", "country", "callback_url"],
    },
    [Target.Tiktok]: {
        group: "TikTok",
        response_format: "html",
        parameters: ["url", "headless", "user_agent_type", "callback_url"],
    },
    [Target.RedditPost]: {
        group: "Reddit",
        response_format: "json",
        parameters: ["url", "locale", "geo", "callback_url"],
    },
    [Target.RedditSubreddit]: {
        group: "Reddit",
        response_format: "json",
        parameters: ["url", "locale", "geo", "callback_url"],
    },
    [Target.RedditUser]: {
        group: "Reddit",
        response_format: "json",
        parameters: ["url", "locale", "geo", "sort", "callback_url"],
    },
    [Target.YoutubeVideo]: {
        group: "None",
        response_format: "json",
        parameters: ["query", "geo", "callback_url"],
    },
    [Target.YoutubeMetadata]: {
        group: "YouTube",
        response_format: "json",
        parameters: ["query", "callback_url"],
    },
    [Target.YoutubeSearch]: {
        group: "YouTube",
        response_format: "json",
        parameters: ["360", "query", "upload_date", "type", "duration", "video_sort_by", "3d", "4k", "creative_commons", "hd", "hdr", "vr180", "live", "location", "purchased", "subtitles", "callback_url"],
    },
    [Target.YoutubeSearchMax]: {
        group: "YouTube",
        response_format: "json",
        parameters: ["360", "query", "upload_date", "type", "duration", "video_sort_by", "3d", "4k", "creative_commons", "hd", "hdr", "vr180", "live", "location", "purchased", "subtitles", "markdown", "callback_url"],
    },
    [Target.YoutubeSubtitles]: {
        group: "YouTube",
        response_format: "json",
        parameters: ["query", "language_code", "subtitle_origin", "callback_url"],
    },
    [Target.YoutubeChannel]: {
        group: "YouTube",
        response_format: "json",
        parameters: ["query", "parse", "limit", "markdown", "callback_url"],
    },
};
