// @decodo/sdk-ts/build/esm/api/web-scraping-api.js
import { prettifyError } from 'zod';
import { ValidationError } from '../errors.js';
import { BundledSchema } from '../schema/bundled-schema.js';
export class WebScrapingApi {
    http;
    schemas;
    constructor(http, schemas = BundledSchema.shared) {
        this.http = http;
        this.schemas = schemas;
    }
    validate(params) {
        const schema = this.schemas.getRequestSchema(params.target);
        if (!schema) {
            return;
        }
        const parsed = schema.safeParse(params);
        if (!parsed.success) {
            throw new ValidationError(prettifyError(parsed.error));
        }
    }
    async scrape(params) {
        this.validate(params);
        return this.http.post('/v2/scrape', params);
    }
    async scrapeAsync(params) {
        this.validate(params);
        return this.http.post('/v3/task', params);
    }
    async scrapeBatch(params) {
        return this.http.post('/v3/task/batch', params);
    }
    async getStatus(taskId) {
        return this.http.get(`/v3/task/${taskId}`);
    }
    async getResults(taskId) {
        const res = await this.http.get(`/v3/task/${taskId}/results`);
        return res ?? null;
    }
}
