// @decodo/sdk-ts/build/esm/schema/remote-schema.js
import { mkdirSync, readFileSync, writeFileSync } from 'node:fs';
import { dirname } from 'node:path';
import * as z from 'zod';
import { buildTargetMeta } from './build-target-meta.js';
import { DEFAULT_IR_CACHE_PATH } from './constants.js';
import { expandPath } from './expand-path.js';
import { resolveLatestIr } from './resolve-latest-ir.js';
const readCachedIr = (cachePath) => {
    try {
        const raw = readFileSync(cachePath, 'utf-8');
        return JSON.parse(raw);
    }
    catch {
        return null;
    }
};
const isCacheFresh = (cached, ttlMs) => {
    if (ttlMs === undefined) {
        return true;
    }
    return Date.now() - cached.fetchedAt < ttlMs;
};
const fetchIr = async (url) => {
    const res = await fetch(url, {
        headers: { Accept: 'application/json' },
    });
    if (!res.ok) {
        throw new Error(`Failed to fetch IR from ${url}: HTTP ${res.status}`);
    }
    return (await res.json());
};
const writeCachedIr = (cachePath, cached) => {
    mkdirSync(dirname(cachePath), { recursive: true });
    writeFileSync(cachePath, JSON.stringify(cached, null, 2));
};
const resolveIrUrl = async (config) => {
    if (config.url) {
        return { url: config.url };
    }
    const latest = await resolveLatestIr();
    return { url: latest.url, version: latest.version };
};
const loadIr = async (config) => {
    const cachePath = expandPath(config.cachePath ?? DEFAULT_IR_CACHE_PATH);
    const cached = readCachedIr(cachePath);
    const resolved = await resolveIrUrl(config);
    if (cached?.resolvedUrl === resolved.url) {
        if (isCacheFresh(cached, config.ttlMs)) {
            return cached.ir;
        }
        if (config.url || cached.ir.version === resolved.version) {
            return cached.ir;
        }
    }
    const ir = await fetchIr(resolved.url);
    writeCachedIr(cachePath, {
        fetchedAt: Date.now(),
        resolvedUrl: resolved.url,
        ir,
    });
    return ir;
};
const buildRequestSchemas = (ir) => {
    const schemas = new Map();
    for (const [targetKey, target] of Object.entries(ir.apis.webScrapingApi.targets)) {
        schemas.set(targetKey, z.fromJSONSchema(target.parameter_schema));
    }
    return schemas;
};
const buildParameterSchemas = (ir) => {
    const schemas = new Map();
    for (const [targetKey, target] of Object.entries(ir.apis.webScrapingApi.targets)) {
        schemas.set(targetKey, target.parameter_schema);
    }
    return schemas;
};
export class RemoteSchema {
    version;
    schemas;
    meta;
    parameterSchemas;
    sharedParameters;
    targetKeys;
    constructor(ir, schemas) {
        this.version = ir.version;
        this.schemas = schemas;
        this.meta = buildTargetMeta(ir.apis.webScrapingApi.targets);
        this.parameterSchemas = buildParameterSchemas(ir);
        this.sharedParameters = ir.apis.webScrapingApi.parameters ?? {};
        this.targetKeys = Object.keys(ir.apis.webScrapingApi.targets);
    }
    static async load(config = {}) {
        const ir = await loadIr(config);
        const schemas = buildRequestSchemas(ir);
        return new RemoteSchema(ir, schemas);
    }
    getRequestSchema(target) {
        return this.schemas.get(target);
    }
    listTargets() {
        return this.targetKeys;
    }
    getTargetMeta(target) {
        return this.meta.get(target);
    }
    getTargetParameterSchema(target) {
        return this.parameterSchemas.get(target);
    }
    getSharedParameters() {
        return this.sharedParameters;
    }
}
