// @decodo/sdk-ts/build/esm/client.js
import { HttpClient } from './http.js';
import { WebScrapingApi } from './api/web-scraping-api.js';
import { BundledSchema } from './schema/bundled-schema.js';
const WEB_API_BASE_URL = 'https://scraper-api.decodo.com';
const DEFAULT_TIMEOUT_MS = 180_000;
const notConfigured = (namespace, hint) => {
    throw new Error(`${namespace} is not configured. ${hint}`);
};
export class DecodoClient {
    webScrapingApi;
    constructor(config) {
        const timeoutMs = config.timeoutMs ?? DEFAULT_TIMEOUT_MS;
        const schema = config.schema ?? BundledSchema.shared;
        if (config.webScrapingApi) {
            this.webScrapingApi = new WebScrapingApi(new HttpClient({
                baseUrl: WEB_API_BASE_URL,
                auth: {
                    type: 'basic',
                    token: config.webScrapingApi.token,
                },
                timeoutMs,
                integrationHeader: config.webScrapingApi.integrationHeader,
            }), schema);
        }
        else {
            this.webScrapingApi = new Proxy({}, {
                get() {
                    notConfigured('webScrapingApi', 'Provide webScrapingApi.token in DecodoConfig.');
                },
            });
        }
    }
}
