// @decodo/sdk-ts/build/esm/errors.js
export class DecodoError extends Error {
    statusCode;
    apiStatus;
    constructor(message, statusCode, apiStatus) {
        super(message);
        this.name = 'DecodoError';
        this.statusCode = statusCode;
        this.apiStatus = apiStatus;
    }
}
export class AuthenticationError extends DecodoError {
    constructor(message = 'Authentication failed. Check your credentials.') {
        super(message, 401, 'failed');
        this.name = 'AuthenticationError';
    }
}
export class RateLimitError extends DecodoError {
    constructor(message = 'Rate limit exceeded. Slow down your request rate.') {
        super(message, 429, 'failed');
        this.name = 'RateLimitError';
    }
}
export class ValidationError extends DecodoError {
    errors;
    constructor(message, errors) {
        super(message, 422, 'failed');
        this.name = 'ValidationError';
        this.errors = errors;
    }
}
export class TimeoutError extends Error {
    constructor(message = 'The request timed out.') {
        super(message);
        this.name = 'TimeoutError';
    }
}
