// @decodo/sdk-ts/build/esm/http.js
import { DecodoError, AuthenticationError, RateLimitError, ValidationError, TimeoutError, } from './errors.js';
export class HttpClient {
    baseUrl;
    authHeader;
    timeoutMs;
    integrationHeader;
    constructor(config) {
        this.baseUrl = config.baseUrl.replace(/\/+$/, '');
        this.timeoutMs = config.timeoutMs;
        this.integrationHeader = config.integrationHeader ?? 'sdk-ts';
        if (config.auth.type === 'basic') {
            this.authHeader = `Basic ${config.auth.token}`;
        }
        else {
            this.authHeader = config.auth.apiKey;
        }
    }
    async request(method, path, body) {
        const url = `${this.baseUrl}${path}`;
        const controller = new AbortController();
        const timer = setTimeout(() => controller.abort(), this.timeoutMs);
        try {
            const init = {
                method,
                headers: {
                    Authorization: this.authHeader,
                    'Content-Type': 'application/json',
                    Accept: 'application/json',
                    'x-integration': this.integrationHeader,
                },
                signal: controller.signal,
            };
            if (body !== undefined) {
                init.body = JSON.stringify(body);
            }
            const res = await fetch(url, init);
            if (res.ok) {
                if (res.status === 204) {
                    return undefined;
                }
                return (await res.json());
            }
            let errorBody;
            try {
                errorBody = (await res.json());
            }
            catch {
                // response body wasn't JSON
            }
            const message = errorBody?.message ?? `HTTP ${res.status}`;
            if (res.status === 401 || res.status === 403) {
                throw new AuthenticationError(message);
            }
            if (res.status === 429) {
                throw new RateLimitError(message);
            }
            if (res.status === 422 || (res.status === 400 && errorBody?.errors)) {
                throw new ValidationError(message, errorBody?.errors);
            }
            throw new DecodoError(message, res.status, errorBody?.status);
        }
        catch (err) {
            if (err instanceof DecodoError) {
                throw err;
            }
            if (err instanceof TypeError &&
                err.cause?.code ===
                    'ABORT_ERR') {
                throw new TimeoutError(`Request to ${path} timed out after ${this.timeoutMs}ms`);
            }
            if (err instanceof DOMException && err.name === 'AbortError') {
                throw new TimeoutError(`Request to ${path} timed out after ${this.timeoutMs}ms`);
            }
            throw err;
        }
        finally {
            clearTimeout(timer);
        }
    }
    post(path, body) {
        return this.request('POST', path, body);
    }
    get(path, query) {
        if (query) {
            const params = new URLSearchParams(query);
            const qs = params.toString();
            if (qs) {
                return this.request('GET', `${path}?${qs}`);
            }
        }
        return this.request('GET', path);
    }
}
