// @decodo/sdk-ts/build/cjs/index.cjs
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var index_exports = {};
__export(index_exports, {
  AuthenticationError: () => AuthenticationError,
  BundledSchema: () => BundledSchema,
  DecodoClient: () => DecodoClient,
  DecodoError: () => DecodoError,
  RateLimitError: () => RateLimitError,
  RemoteSchema: () => RemoteSchema,
  Target: () => Target,
  TimeoutError: () => TimeoutError,
  ValidationError: () => ValidationError,
  WebScrapingApi: () => WebScrapingApi,
  parameterMeta: () => parameterMeta,
  targetMeta: () => targetMeta,
  targets: () => targets
});
module.exports = __toCommonJS(index_exports);

// src/errors.ts
var DecodoError = class extends Error {
  statusCode;
  apiStatus;
  constructor(message, statusCode, apiStatus) {
    super(message);
    this.name = "DecodoError";
    this.statusCode = statusCode;
    this.apiStatus = apiStatus;
  }
};
var AuthenticationError = class extends DecodoError {
  constructor(message = "Authentication failed. Check your credentials.") {
    super(message, 401, "failed");
    this.name = "AuthenticationError";
  }
};
var RateLimitError = class extends DecodoError {
  constructor(message = "Rate limit exceeded. Slow down your request rate.") {
    super(message, 429, "failed");
    this.name = "RateLimitError";
  }
};
var ValidationError = class extends DecodoError {
  errors;
  constructor(message, errors) {
    super(message, 422, "failed");
    this.name = "ValidationError";
    this.errors = errors;
  }
};
var TimeoutError = class extends Error {
  constructor(message = "The request timed out.") {
    super(message);
    this.name = "TimeoutError";
  }
};

// src/http.ts
var HttpClient = class {
  baseUrl;
  authHeader;
  timeoutMs;
  integrationHeader;
  constructor(config) {
    this.baseUrl = config.baseUrl.replace(/\/+$/, "");
    this.timeoutMs = config.timeoutMs;
    this.integrationHeader = config.integrationHeader ?? "sdk-ts";
    if (config.auth.type === "basic") {
      this.authHeader = `Basic ${config.auth.token}`;
    } else {
      this.authHeader = config.auth.apiKey;
    }
  }
  async request(method, path, body) {
    const url = `${this.baseUrl}${path}`;
    const controller = new AbortController();
    const timer = setTimeout(() => controller.abort(), this.timeoutMs);
    try {
      const init = {
        method,
        headers: {
          Authorization: this.authHeader,
          "Content-Type": "application/json",
          Accept: "application/json",
          "x-integration": this.integrationHeader
        },
        signal: controller.signal
      };
      if (body !== void 0) {
        init.body = JSON.stringify(body);
      }
      const res = await fetch(url, init);
      if (res.ok) {
        if (res.status === 204) {
          return void 0;
        }
        return await res.json();
      }
      let errorBody;
      try {
        errorBody = await res.json();
      } catch {
      }
      const message = errorBody?.message ?? `HTTP ${res.status}`;
      if (res.status === 401 || res.status === 403) {
        throw new AuthenticationError(message);
      }
      if (res.status === 429) {
        throw new RateLimitError(message);
      }
      if (res.status === 422 || res.status === 400 && errorBody?.errors) {
        throw new ValidationError(message, errorBody?.errors);
      }
      throw new DecodoError(message, res.status, errorBody?.status);
    } catch (err) {
      if (err instanceof DecodoError) {
        throw err;
      }
      if (err instanceof TypeError && err.cause?.code === "ABORT_ERR") {
        throw new TimeoutError(
          `Request to ${path} timed out after ${this.timeoutMs}ms`
        );
      }
      if (err instanceof DOMException && err.name === "AbortError") {
        throw new TimeoutError(
          `Request to ${path} timed out after ${this.timeoutMs}ms`
        );
      }
      throw err;
    } finally {
      clearTimeout(timer);
    }
  }
  post(path, body) {
    return this.request("POST", path, body);
  }
  get(path, query) {
    if (query) {
      const params = new URLSearchParams(query);
      const qs = params.toString();
      if (qs) {
        return this.request("GET", `${path}?${qs}`);
      }
    }
    return this.request("GET", path);
  }
};

// src/api/web-scraping-api.ts
var import_zod = require("zod");

// src/generated/request-schemas.ts
var z = __toESM(require("zod"), 1);

// src/generated/targets.ts
var Target = /* @__PURE__ */ ((_Target) => {
  _Target["UniversalEcommerce"] = "universal_ecommerce";
  _Target["GoogleSearch"] = "google_search";
  _Target["GoogleTravelHotels"] = "google_travel_hotels";
  _Target["GoogleTrendsExplore"] = "google_trends_explore";
  _Target["GoogleShoppingSearch"] = "google_shopping_search";
  _Target["GoogleShoppingProduct"] = "google_shopping_product";
  _Target["Google"] = "google";
  _Target["GoogleSuggest"] = "google_suggest";
  _Target["GoogleMaps"] = "google_maps";
  _Target["GoogleAiMode"] = "google_ai_mode";
  _Target["GoogleAds"] = "google_ads";
  _Target["GoogleLens"] = "google_lens";
  _Target["BingSearch"] = "bing_search";
  _Target["Bing"] = "bing";
  _Target["YoutubeTranscript"] = "youtube_transcript";
  _Target["AmazonProduct"] = "amazon_product";
  _Target["AmazonPricing"] = "amazon_pricing";
  _Target["AmazonSearch"] = "amazon_search";
  _Target["AmazonSellers"] = "amazon_sellers";
  _Target["AmazonBestsellers"] = "amazon_bestsellers";
  _Target["Amazon"] = "amazon";
  _Target["Ecommerce"] = "ecommerce";
  _Target["WalmartProduct"] = "walmart_product";
  _Target["WalmartSearch"] = "walmart_search";
  _Target["Walmart"] = "walmart";
  _Target["TargetProduct"] = "target_product";
  _Target["TargetSearch"] = "target_search";
  _Target["Target"] = "target";
  _Target["LowesSearch"] = "lowes_search";
  _Target["Universal"] = "universal";
  _Target["Chatgpt"] = "chatgpt";
  _Target["Perplexity"] = "perplexity";
  _Target["Gemini"] = "gemini";
  _Target["Bbb"] = "bbb";
  _Target["Autotrader"] = "autotrader";
  _Target["Mobile"] = "mobile";
  _Target["Airbnb"] = "airbnb";
  _Target["AppleAppStore"] = "apple_app_store";
  _Target["InstagramGraphqlProfile"] = "instagram_graphql_profile";
  _Target["TiktokPost"] = "tiktok_post";
  _Target["TiktokShopSearch"] = "tiktok_shop_search";
  _Target["TiktokShopProduct"] = "tiktok_shop_product";
  _Target["Tiktok"] = "tiktok";
  _Target["RedditPost"] = "reddit_post";
  _Target["RedditSubreddit"] = "reddit_subreddit";
  _Target["RedditUser"] = "reddit_user";
  _Target["YoutubeVideo"] = "youtube_video";
  _Target["YoutubeMetadata"] = "youtube_metadata";
  _Target["YoutubeSearch"] = "youtube_search";
  _Target["YoutubeSearchMax"] = "youtube_search_max";
  _Target["YoutubeSubtitles"] = "youtube_subtitles";
  _Target["YoutubeChannel"] = "youtube_channel";
  return _Target;
})(Target || {});
var targets = ["universal_ecommerce" /* UniversalEcommerce */, "google_search" /* GoogleSearch */, "google_travel_hotels" /* GoogleTravelHotels */, "google_trends_explore" /* GoogleTrendsExplore */, "google_shopping_search" /* GoogleShoppingSearch */, "google_shopping_product" /* GoogleShoppingProduct */, "google" /* Google */, "google_suggest" /* GoogleSuggest */, "google_maps" /* GoogleMaps */, "google_ai_mode" /* GoogleAiMode */, "google_ads" /* GoogleAds */, "google_lens" /* GoogleLens */, "bing_search" /* BingSearch */, "bing" /* Bing */, "youtube_transcript" /* YoutubeTranscript */, "amazon_product" /* AmazonProduct */, "amazon_pricing" /* AmazonPricing */, "amazon_search" /* AmazonSearch */, "amazon_sellers" /* AmazonSellers */, "amazon_bestsellers" /* AmazonBestsellers */, "amazon" /* Amazon */, "ecommerce" /* Ecommerce */, "walmart_product" /* WalmartProduct */, "walmart_search" /* WalmartSearch */, "walmart" /* Walmart */, "target_product" /* TargetProduct */, "target_search" /* TargetSearch */, "target" /* Target */, "lowes_search" /* LowesSearch */, "universal" /* Universal */, "chatgpt" /* Chatgpt */, "perplexity" /* Perplexity */, "gemini" /* Gemini */, "bbb" /* Bbb */, "autotrader" /* Autotrader */, "mobile" /* Mobile */, "airbnb" /* Airbnb */, "apple_app_store" /* AppleAppStore */, "instagram_graphql_profile" /* InstagramGraphqlProfile */, "tiktok_post" /* TiktokPost */, "tiktok_shop_search" /* TiktokShopSearch */, "tiktok_shop_product" /* TiktokShopProduct */, "tiktok" /* Tiktok */, "reddit_post" /* RedditPost */, "reddit_subreddit" /* RedditSubreddit */, "reddit_user" /* RedditUser */, "youtube_video" /* YoutubeVideo */, "youtube_metadata" /* YoutubeMetadata */, "youtube_search" /* YoutubeSearch */, "youtube_search_max" /* YoutubeSearchMax */, "youtube_subtitles" /* YoutubeSubtitles */, "youtube_channel" /* YoutubeChannel */];
var targetMeta = {
  ["universal_ecommerce" /* UniversalEcommerce */]: {
    group: "None",
    response_format: "html",
    parameters: ["callback_url"]
  },
  ["google_search" /* GoogleSearch */]: {
    group: "Google",
    response_format: "json",
    parameters: ["query", "headless", "locale", "geo", "device_type", "page_from", "google_results_language", "google_tbm", "google_tbs", "parse", "google_nfpr", "google_safe_search", "session_id", "xhr", "markdown", "page_count", "callback_url"]
  },
  ["google_travel_hotels" /* GoogleTravelHotels */]: {
    group: "Google",
    response_format: "html",
    parameters: ["query", "headless", "locale", "device_type", "page_from", "date_range", "stars", "adults", "children", "session_id", "markdown", "callback_url"]
  },
  ["google_trends_explore" /* GoogleTrendsExplore */]: {
    group: "Google",
    response_format: "json",
    parameters: ["query", "geo", "device_type", "search_type", "date_start", "date_end", "callback_url"]
  },
  ["google_shopping_search" /* GoogleShoppingSearch */]: {
    group: "Google",
    response_format: "json",
    parameters: ["query", "headless", "locale", "geo", "device_type", "page_from", "google_tbs", "parse", "session_id", "google_results_language", "markdown", "callback_url"]
  },
  ["google_shopping_product" /* GoogleShoppingProduct */]: {
    group: "Google",
    response_format: "json",
    parameters: ["query", "headless", "locale", "geo", "device_type", "page_from", "parse", "session_id", "google_results_language", "markdown", "xhr", "callback_url"]
  },
  ["google" /* Google */]: {
    group: "Google",
    response_format: "json",
    parameters: ["url", "headless", "locale", "device_type", "parse", "session_id", "markdown", "xhr", "page_count", "callback_url"]
  },
  ["google_suggest" /* GoogleSuggest */]: {
    group: "Google",
    response_format: "json",
    parameters: ["query", "device_type", "geo", "locale", "session_id", "callback_url"]
  },
  ["google_maps" /* GoogleMaps */]: {
    group: "Google",
    response_format: "html",
    parameters: ["query", "headless", "geo", "locale", "page_from", "device_type", "session_id", "google_results_language", "google_nfpr", "hotel_occupancy", "date_range", "markdown", "callback_url"]
  },
  ["google_ai_mode" /* GoogleAiMode */]: {
    group: "AI Tools",
    response_format: "json",
    parameters: ["query", "geo", "parse", "device_type", "session_id", "markdown", "xhr", "callback_url"]
  },
  ["google_ads" /* GoogleAds */]: {
    group: "Google",
    response_format: "json",
    parameters: ["query", "headless", "locale", "geo", "device_type", "page_from", "google_results_language", "google_tbm", "google_tbs", "parse", "google_nfpr", "session_id", "markdown", "xhr", "page_count", "callback_url"]
  },
  ["google_lens" /* GoogleLens */]: {
    group: "Google",
    response_format: "json",
    parameters: ["query", "headless", "parse", "device_type", "markdown", "callback_url"]
  },
  ["bing_search" /* BingSearch */]: {
    group: "Bing",
    response_format: "json",
    parameters: ["query", "headless", "locale", "geo", "domain", "device_type", "page_from", "parse", "page_count", "session_id", "markdown", "xhr", "callback_url"]
  },
  ["bing" /* Bing */]: {
    group: "Bing",
    response_format: "json",
    parameters: ["url", "headless", "locale", "geo", "device_type", "page_from", "parse", "session_id", "markdown", "xhr", "callback_url"]
  },
  ["youtube_transcript" /* YoutubeTranscript */]: {
    group: "YouTube",
    response_format: "json",
    parameters: ["query", "language_code", "transcript_origin", "callback_url"]
  },
  ["amazon_product" /* AmazonProduct */]: {
    group: "Amazon",
    response_format: "json",
    parameters: ["query", "headless", "domain", "device_type", "parse", "autoselect_variant", "geo", "session_id", "currency", "markdown", "xhr", "callback_url"]
  },
  ["amazon_pricing" /* AmazonPricing */]: {
    group: "Amazon",
    response_format: "json",
    parameters: ["query", "headless", "domain", "device_type", "page_from", "parse", "geo", "session_id", "currency", "markdown", "xhr", "callback_url"]
  },
  ["amazon_search" /* AmazonSearch */]: {
    group: "Amazon",
    response_format: "json",
    parameters: ["query", "headless", "domain", "device_type", "page_from", "category", "merchant", "parse", "geo", "session_id", "sort_by", "currency", "markdown", "xhr", "callback_url"]
  },
  ["amazon_sellers" /* AmazonSellers */]: {
    group: "Amazon",
    response_format: "json",
    parameters: ["query", "headless", "locale", "domain", "device_type", "geo", "parse", "markdown", "xhr", "callback_url"]
  },
  ["amazon_bestsellers" /* AmazonBestsellers */]: {
    group: "Amazon",
    response_format: "json",
    parameters: ["query", "domain", "device_type", "geo", "page_from", "category", "parse", "session_id", "currency", "markdown", "xhr", "callback_url"]
  },
  ["amazon" /* Amazon */]: {
    group: "Amazon",
    response_format: "json",
    parameters: ["url", "headless", "device_type", "parse", "geo", "session_id", "markdown", "xhr", "callback_url"]
  },
  ["ecommerce" /* Ecommerce */]: {
    group: "Other eCommerce",
    response_format: "json",
    parameters: ["url", "headless", "locale", "geo", "device_type", "parse", "parser_type", "callback_url"]
  },
  ["walmart_product" /* WalmartProduct */]: {
    group: "Walmart",
    response_format: "html",
    parameters: ["product_id", "headless", "parse", "xhr", "markdown", "fulfillment_type", "walmart_store_id", "delivery_zip", "callback_url"]
  },
  ["walmart_search" /* WalmartSearch */]: {
    group: "Walmart",
    response_format: "json",
    parameters: ["query", "headless", "parse", "markdown", "fulfillment_type", "walmart_store_id", "delivery_zip", "callback_url"]
  },
  ["walmart" /* Walmart */]: {
    group: "Walmart",
    response_format: "html",
    parameters: ["url", "headless", "locale", "geo", "device_type", "store_id", "markdown", "xhr", "callback_url"]
  },
  ["target_product" /* TargetProduct */]: {
    group: "Target",
    response_format: "json",
    parameters: ["product_id", "headless", "parse", "device_type", "markdown", "xhr", "delivery_type", "target_store_id", "delivery_zip", "callback_url"]
  },
  ["target_search" /* TargetSearch */]: {
    group: "Target",
    response_format: "json",
    parameters: ["query", "headless", "parse", "device_type", "delivery_type", "target_store_id", "delivery_zip", "xhr", "markdown", "callback_url"]
  },
  ["target" /* Target */]: {
    group: "Target",
    response_format: "html",
    parameters: ["url", "headless", "device_type", "xhr", "delivery_zip", "target_store_id", "callback_url"]
  },
  ["lowes_search" /* LowesSearch */]: {
    group: "Lowe's",
    response_format: "json",
    parameters: ["query", "lowes_store_id", "headless", "delivery_zip", "user_agent_type", "free_delivery", "pickup_today", "delivery_today_tomorrow", "callback_url"]
  },
  ["universal" /* Universal */]: {
    group: "Universal",
    response_format: "html",
    parameters: ["url", "payload", "proxy_pool", "http_method", "headless", "geo", "locale", "device_type", "session_id", "successful_status_codes", "headers", "cookies", "force_headers", "force_cookies", "xhr", "markdown", "callback_url"]
  },
  ["chatgpt" /* Chatgpt */]: {
    group: "AI Tools",
    response_format: "json",
    parameters: ["prompt", "search", "parse", "geo", "device_type", "markdown", "xhr", "callback_url"]
  },
  ["perplexity" /* Perplexity */]: {
    group: "AI Tools",
    response_format: "json",
    parameters: ["prompt", "parse", "geo", "device_type", "markdown", "xhr", "callback_url"]
  },
  ["gemini" /* Gemini */]: {
    group: "AI Tools",
    response_format: "json",
    parameters: ["prompt", "parse", "geo", "xhr", "callback_url"]
  },
  ["bbb" /* Bbb */]: {
    group: "Business Reviews",
    response_format: "html",
    parameters: ["url", "headless", "geo", "device_type", "markdown", "xhr", "callback_url"]
  },
  ["autotrader" /* Autotrader */]: {
    group: "Marketplace",
    response_format: "html",
    parameters: ["url", "headless", "geo", "device_type", "markdown", "xhr", "callback_url"]
  },
  ["mobile" /* Mobile */]: {
    group: "Marketplace",
    response_format: "html",
    parameters: ["url", "headless", "geo", "device_type", "markdown", "xhr", "callback_url"]
  },
  ["airbnb" /* Airbnb */]: {
    group: "Travel",
    response_format: "html",
    parameters: ["url", "headless", "geo", "device_type", "markdown", "xhr", "callback_url"]
  },
  ["apple_app_store" /* AppleAppStore */]: {
    group: "Marketplace",
    response_format: "html",
    parameters: ["url", "headless", "geo", "device_type", "markdown", "xhr", "callback_url"]
  },
  ["instagram_graphql_profile" /* InstagramGraphqlProfile */]: {
    group: "Instagram",
    response_format: "json",
    parameters: ["query", "callback_url"]
  },
  ["tiktok_post" /* TiktokPost */]: {
    group: "TikTok",
    response_format: "html",
    parameters: ["url", "xhr", "callback_url"]
  },
  ["tiktok_shop_search" /* TiktokShopSearch */]: {
    group: "TikTok",
    response_format: "html",
    parameters: ["query", "headless", "device_type", "markdown", "country", "callback_url"]
  },
  ["tiktok_shop_product" /* TiktokShopProduct */]: {
    group: "TikTok",
    response_format: "html",
    parameters: ["product_id", "headless", "device_type", "xhr", "markdown", "country", "callback_url"]
  },
  ["tiktok" /* Tiktok */]: {
    group: "TikTok",
    response_format: "html",
    parameters: ["url", "headless", "user_agent_type", "callback_url"]
  },
  ["reddit_post" /* RedditPost */]: {
    group: "Reddit",
    response_format: "json",
    parameters: ["url", "locale", "geo", "callback_url"]
  },
  ["reddit_subreddit" /* RedditSubreddit */]: {
    group: "Reddit",
    response_format: "json",
    parameters: ["url", "locale", "geo", "callback_url"]
  },
  ["reddit_user" /* RedditUser */]: {
    group: "Reddit",
    response_format: "json",
    parameters: ["url", "locale", "geo", "sort", "callback_url"]
  },
  ["youtube_video" /* YoutubeVideo */]: {
    group: "None",
    response_format: "json",
    parameters: ["query", "geo", "callback_url"]
  },
  ["youtube_metadata" /* YoutubeMetadata */]: {
    group: "YouTube",
    response_format: "json",
    parameters: ["query", "callback_url"]
  },
  ["youtube_search" /* YoutubeSearch */]: {
    group: "YouTube",
    response_format: "json",
    parameters: ["360", "query", "upload_date", "type", "duration", "video_sort_by", "3d", "4k", "creative_commons", "hd", "hdr", "vr180", "live", "location", "purchased", "subtitles", "callback_url"]
  },
  ["youtube_search_max" /* YoutubeSearchMax */]: {
    group: "YouTube",
    response_format: "json",
    parameters: ["360", "query", "upload_date", "type", "duration", "video_sort_by", "3d", "4k", "creative_commons", "hd", "hdr", "vr180", "live", "location", "purchased", "subtitles", "markdown", "callback_url"]
  },
  ["youtube_subtitles" /* YoutubeSubtitles */]: {
    group: "YouTube",
    response_format: "json",
    parameters: ["query", "language_code", "subtitle_origin", "callback_url"]
  },
  ["youtube_channel" /* YoutubeChannel */]: {
    group: "YouTube",
    response_format: "json",
    parameters: ["query", "parse", "limit", "markdown", "callback_url"]
  }
};

// src/generated/request-schemas.ts
var requestJsonSchemas = {
  ["universal_ecommerce" /* UniversalEcommerce */]: {
    "type": "object",
    "properties": {
      "target": {
        "const": "universal_ecommerce"
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["google_search" /* GoogleSearch */]: {
    "type": "object",
    "properties": {
      "target": {
        "const": "google_search"
      },
      "query": {
        "type": "string",
        "maxLength": 2048
      },
      "headless": {
        "type": "string",
        "enum": [
          "html",
          "png"
        ]
      },
      "locale": {
        "type": "string"
      },
      "geo": {
        "type": "string"
      },
      "device_type": {
        "type": "string"
      },
      "page_from": {
        "type": "number",
        "minimum": 0,
        "maximum": 100
      },
      "google_results_language": {
        "type": "string"
      },
      "google_tbm": {
        "type": "string"
      },
      "google_tbs": {
        "type": "string"
      },
      "parse": {
        "type": "boolean"
      },
      "google_nfpr": {
        "type": "boolean"
      },
      "google_safe_search": {
        "type": "boolean"
      },
      "session_id": {
        "type": "string"
      },
      "xhr": {
        "type": "boolean"
      },
      "markdown": {
        "type": "boolean"
      },
      "page_count": {
        "type": "number",
        "minimum": 1,
        "maximum": 10
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["google_travel_hotels" /* GoogleTravelHotels */]: {
    "type": "object",
    "properties": {
      "target": {
        "const": "google_travel_hotels"
      },
      "query": {
        "type": "string",
        "maxLength": 2048
      },
      "headless": {
        "type": "string",
        "enum": [
          "html",
          "png"
        ]
      },
      "locale": {
        "type": "string"
      },
      "device_type": {
        "type": "string"
      },
      "page_from": {
        "type": "number",
        "minimum": 0,
        "maximum": 100
      },
      "date_range": {
        "type": "string"
      },
      "stars": {
        "type": "number"
      },
      "adults": {
        "type": "number",
        "minimum": 1,
        "maximum": 3
      },
      "children": {
        "type": "number",
        "minimum": 1,
        "maximum": 3
      },
      "session_id": {
        "type": "string"
      },
      "markdown": {
        "type": "boolean"
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["google_trends_explore" /* GoogleTrendsExplore */]: {
    "type": "object",
    "properties": {
      "target": {
        "const": "google_trends_explore"
      },
      "query": {
        "type": "string",
        "maxLength": 2048
      },
      "geo": {
        "type": "string"
      },
      "device_type": {
        "type": "string"
      },
      "search_type": {
        "type": "string"
      },
      "date_start": {
        "type": "string"
      },
      "date_end": {
        "type": "string"
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["google_shopping_search" /* GoogleShoppingSearch */]: {
    "type": "object",
    "properties": {
      "target": {
        "const": "google_shopping_search"
      },
      "query": {
        "type": "string",
        "maxLength": 2048
      },
      "headless": {
        "type": "string",
        "enum": [
          "html",
          "png"
        ]
      },
      "locale": {
        "type": "string"
      },
      "geo": {
        "type": "string"
      },
      "device_type": {
        "type": "string"
      },
      "page_from": {
        "type": "number",
        "minimum": 0,
        "maximum": 100
      },
      "google_tbs": {
        "type": "string"
      },
      "parse": {
        "type": "boolean"
      },
      "session_id": {
        "type": "string"
      },
      "google_results_language": {
        "type": "string"
      },
      "markdown": {
        "type": "boolean"
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["google_shopping_product" /* GoogleShoppingProduct */]: {
    "type": "object",
    "properties": {
      "target": {
        "const": "google_shopping_product"
      },
      "query": {
        "type": "string",
        "maxLength": 2048
      },
      "headless": {
        "type": "string",
        "enum": [
          "html",
          "png"
        ]
      },
      "locale": {
        "type": "string"
      },
      "geo": {
        "type": "string"
      },
      "device_type": {
        "type": "string"
      },
      "page_from": {
        "type": "number",
        "minimum": 0,
        "maximum": 100
      },
      "parse": {
        "type": "boolean"
      },
      "session_id": {
        "type": "string"
      },
      "google_results_language": {
        "type": "string"
      },
      "markdown": {
        "type": "boolean"
      },
      "xhr": {
        "type": "boolean"
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["google" /* Google */]: {
    "type": "object",
    "properties": {
      "target": {
        "const": "google"
      },
      "url": {
        "type": "string",
        "maxLength": 2056
      },
      "headless": {
        "type": "string",
        "enum": [
          "html",
          "png"
        ]
      },
      "locale": {
        "type": "string"
      },
      "device_type": {
        "type": "string"
      },
      "parse": {
        "type": "boolean"
      },
      "session_id": {
        "type": "string"
      },
      "markdown": {
        "type": "boolean"
      },
      "xhr": {
        "type": "boolean"
      },
      "page_count": {
        "type": "number",
        "minimum": 1,
        "maximum": 10
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["google_suggest" /* GoogleSuggest */]: {
    "type": "object",
    "properties": {
      "target": {
        "const": "google_suggest"
      },
      "query": {
        "type": "string",
        "maxLength": 2048
      },
      "device_type": {
        "type": "string"
      },
      "geo": {
        "type": "string"
      },
      "locale": {
        "type": "string"
      },
      "session_id": {
        "type": "string"
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["google_maps" /* GoogleMaps */]: {
    "type": "object",
    "properties": {
      "target": {
        "const": "google_maps"
      },
      "query": {
        "type": "string",
        "maxLength": 2048
      },
      "headless": {
        "type": "string",
        "enum": [
          "html",
          "png"
        ]
      },
      "geo": {
        "type": "string"
      },
      "locale": {
        "type": "string"
      },
      "page_from": {
        "type": "number",
        "minimum": 0,
        "maximum": 100
      },
      "device_type": {
        "type": "string"
      },
      "session_id": {
        "type": "string"
      },
      "google_results_language": {
        "type": "string"
      },
      "google_nfpr": {
        "type": "boolean"
      },
      "hotel_occupancy": {
        "type": "string"
      },
      "date_range": {
        "type": "string"
      },
      "markdown": {
        "type": "boolean"
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["google_ai_mode" /* GoogleAiMode */]: {
    "type": "object",
    "properties": {
      "target": {
        "const": "google_ai_mode"
      },
      "query": {
        "type": "string",
        "maxLength": 2048
      },
      "geo": {
        "type": "string"
      },
      "parse": {
        "type": "boolean"
      },
      "device_type": {
        "type": "string"
      },
      "session_id": {
        "type": "string"
      },
      "markdown": {
        "type": "boolean"
      },
      "xhr": {
        "type": "boolean"
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["google_ads" /* GoogleAds */]: {
    "type": "object",
    "properties": {
      "target": {
        "const": "google_ads"
      },
      "query": {
        "type": "string",
        "maxLength": 2048
      },
      "headless": {
        "type": "string",
        "enum": [
          "html",
          "png"
        ]
      },
      "locale": {
        "type": "string"
      },
      "geo": {
        "type": "string"
      },
      "device_type": {
        "type": "string"
      },
      "page_from": {
        "type": "number",
        "minimum": 0,
        "maximum": 100
      },
      "google_results_language": {
        "type": "string"
      },
      "google_tbm": {
        "type": "string"
      },
      "google_tbs": {
        "type": "string"
      },
      "parse": {
        "type": "boolean"
      },
      "google_nfpr": {
        "type": "boolean"
      },
      "session_id": {
        "type": "string"
      },
      "markdown": {
        "type": "boolean"
      },
      "xhr": {
        "type": "boolean"
      },
      "page_count": {
        "type": "number",
        "minimum": 1,
        "maximum": 10
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["google_lens" /* GoogleLens */]: {
    "type": "object",
    "properties": {
      "target": {
        "const": "google_lens"
      },
      "query": {
        "type": "string",
        "maxLength": 2048
      },
      "headless": {
        "type": "string",
        "enum": [
          "html",
          "png"
        ]
      },
      "parse": {
        "type": "boolean"
      },
      "device_type": {
        "type": "string"
      },
      "markdown": {
        "type": "boolean"
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["bing_search" /* BingSearch */]: {
    "type": "object",
    "properties": {
      "target": {
        "const": "bing_search"
      },
      "query": {
        "type": "string",
        "maxLength": 2048
      },
      "headless": {
        "type": "string",
        "enum": [
          "html",
          "png"
        ]
      },
      "locale": {
        "type": "string"
      },
      "geo": {
        "type": "string"
      },
      "domain": {
        "type": "string"
      },
      "device_type": {
        "type": "string"
      },
      "page_from": {
        "type": "number",
        "minimum": 0,
        "maximum": 100
      },
      "parse": {
        "type": "boolean"
      },
      "page_count": {
        "type": "number",
        "minimum": 1,
        "maximum": 10
      },
      "session_id": {
        "type": "string"
      },
      "markdown": {
        "type": "boolean"
      },
      "xhr": {
        "type": "boolean"
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["bing" /* Bing */]: {
    "type": "object",
    "properties": {
      "target": {
        "const": "bing"
      },
      "url": {
        "type": "string",
        "maxLength": 2056
      },
      "headless": {
        "type": "string",
        "enum": [
          "html",
          "png"
        ]
      },
      "locale": {
        "type": "string"
      },
      "geo": {
        "type": "string"
      },
      "device_type": {
        "type": "string"
      },
      "page_from": {
        "type": "number",
        "minimum": 0,
        "maximum": 100
      },
      "parse": {
        "type": "boolean"
      },
      "session_id": {
        "type": "string"
      },
      "markdown": {
        "type": "boolean"
      },
      "xhr": {
        "type": "boolean"
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["youtube_transcript" /* YoutubeTranscript */]: {
    "type": "object",
    "properties": {
      "target": {
        "const": "youtube_transcript"
      },
      "query": {
        "type": "string",
        "maxLength": 2048
      },
      "language_code": {
        "type": "string"
      },
      "transcript_origin": {
        "type": "string"
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["amazon_product" /* AmazonProduct */]: {
    "type": "object",
    "properties": {
      "target": {
        "const": "amazon_product"
      },
      "query": {
        "type": "string",
        "maxLength": 2048
      },
      "headless": {
        "type": "string",
        "enum": [
          "html",
          "png"
        ]
      },
      "domain": {
        "type": "string"
      },
      "device_type": {
        "type": "string"
      },
      "parse": {
        "type": "boolean"
      },
      "autoselect_variant": {
        "type": "boolean"
      },
      "geo": {
        "type": "string"
      },
      "session_id": {
        "type": "string"
      },
      "currency": {
        "type": "string"
      },
      "markdown": {
        "type": "boolean"
      },
      "xhr": {
        "type": "boolean"
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["amazon_pricing" /* AmazonPricing */]: {
    "type": "object",
    "properties": {
      "target": {
        "const": "amazon_pricing"
      },
      "query": {
        "type": "string",
        "maxLength": 2048
      },
      "headless": {
        "type": "string",
        "enum": [
          "html",
          "png"
        ]
      },
      "domain": {
        "type": "string"
      },
      "device_type": {
        "type": "string"
      },
      "page_from": {
        "type": "number",
        "minimum": 0,
        "maximum": 100
      },
      "parse": {
        "type": "boolean"
      },
      "geo": {
        "type": "string"
      },
      "session_id": {
        "type": "string"
      },
      "currency": {
        "type": "string"
      },
      "markdown": {
        "type": "boolean"
      },
      "xhr": {
        "type": "boolean"
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["amazon_search" /* AmazonSearch */]: {
    "type": "object",
    "properties": {
      "target": {
        "const": "amazon_search"
      },
      "query": {
        "type": "string",
        "maxLength": 2048
      },
      "headless": {
        "type": "string",
        "enum": [
          "html",
          "png"
        ]
      },
      "domain": {
        "type": "string"
      },
      "device_type": {
        "type": "string"
      },
      "page_from": {
        "type": "number",
        "minimum": 0,
        "maximum": 100
      },
      "category": {
        "type": "string"
      },
      "merchant": {
        "type": "string"
      },
      "parse": {
        "type": "boolean"
      },
      "geo": {
        "type": "string"
      },
      "session_id": {
        "type": "string"
      },
      "sort_by": {
        "type": "string"
      },
      "currency": {
        "type": "string"
      },
      "markdown": {
        "type": "boolean"
      },
      "xhr": {
        "type": "boolean"
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["amazon_sellers" /* AmazonSellers */]: {
    "type": "object",
    "properties": {
      "target": {
        "const": "amazon_sellers"
      },
      "query": {
        "type": "string",
        "maxLength": 2048
      },
      "headless": {
        "type": "string",
        "enum": [
          "html",
          "png"
        ]
      },
      "locale": {
        "type": "string"
      },
      "domain": {
        "type": "string"
      },
      "device_type": {
        "type": "string"
      },
      "geo": {
        "type": "string"
      },
      "parse": {
        "type": "boolean"
      },
      "markdown": {
        "type": "boolean"
      },
      "xhr": {
        "type": "boolean"
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["amazon_bestsellers" /* AmazonBestsellers */]: {
    "type": "object",
    "properties": {
      "target": {
        "const": "amazon_bestsellers"
      },
      "query": {
        "type": "string",
        "maxLength": 2048
      },
      "domain": {
        "type": "string"
      },
      "device_type": {
        "type": "string"
      },
      "geo": {
        "type": "string"
      },
      "page_from": {
        "type": "number",
        "minimum": 0,
        "maximum": 100
      },
      "category": {
        "type": "string"
      },
      "parse": {
        "type": "boolean"
      },
      "session_id": {
        "type": "string"
      },
      "currency": {
        "type": "string"
      },
      "markdown": {
        "type": "boolean"
      },
      "xhr": {
        "type": "boolean"
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["amazon" /* Amazon */]: {
    "type": "object",
    "properties": {
      "target": {
        "const": "amazon"
      },
      "url": {
        "type": "string",
        "maxLength": 2056
      },
      "headless": {
        "type": "string",
        "enum": [
          "html",
          "png"
        ]
      },
      "device_type": {
        "type": "string"
      },
      "parse": {
        "type": "boolean"
      },
      "geo": {
        "type": "string"
      },
      "session_id": {
        "type": "string"
      },
      "markdown": {
        "type": "boolean"
      },
      "xhr": {
        "type": "boolean"
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["ecommerce" /* Ecommerce */]: {
    "type": "object",
    "properties": {
      "target": {
        "const": "ecommerce"
      },
      "url": {
        "type": "string",
        "maxLength": 2056
      },
      "headless": {
        "type": "string",
        "enum": [
          "html",
          "png"
        ]
      },
      "locale": {
        "type": "string"
      },
      "geo": {
        "type": "string"
      },
      "device_type": {
        "type": "string"
      },
      "parse": {
        "type": "boolean"
      },
      "parser_type": {
        "type": "string"
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["walmart_product" /* WalmartProduct */]: {
    "type": "object",
    "properties": {
      "target": {
        "const": "walmart_product"
      },
      "product_id": {
        "type": "string"
      },
      "headless": {
        "type": "string",
        "enum": [
          "html",
          "png"
        ]
      },
      "parse": {
        "type": "boolean"
      },
      "xhr": {
        "type": "boolean"
      },
      "markdown": {
        "type": "boolean"
      },
      "fulfillment_type": {
        "type": "string"
      },
      "walmart_store_id": {
        "type": "string"
      },
      "delivery_zip": {
        "type": "string"
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["walmart_search" /* WalmartSearch */]: {
    "type": "object",
    "properties": {
      "target": {
        "const": "walmart_search"
      },
      "query": {
        "type": "string",
        "maxLength": 2048
      },
      "headless": {
        "type": "string",
        "enum": [
          "html",
          "png"
        ]
      },
      "parse": {
        "type": "boolean"
      },
      "markdown": {
        "type": "boolean"
      },
      "fulfillment_type": {
        "type": "string"
      },
      "walmart_store_id": {
        "type": "string"
      },
      "delivery_zip": {
        "type": "string"
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["walmart" /* Walmart */]: {
    "type": "object",
    "properties": {
      "target": {
        "const": "walmart"
      },
      "url": {
        "type": "string",
        "maxLength": 2056
      },
      "headless": {
        "type": "string",
        "enum": [
          "html",
          "png"
        ]
      },
      "locale": {
        "type": "string"
      },
      "geo": {
        "type": "string"
      },
      "device_type": {
        "type": "string"
      },
      "store_id": {
        "type": "string"
      },
      "markdown": {
        "type": "boolean"
      },
      "xhr": {
        "type": "boolean"
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["target_product" /* TargetProduct */]: {
    "type": "object",
    "properties": {
      "target": {
        "const": "target_product"
      },
      "product_id": {
        "type": "string"
      },
      "headless": {
        "type": "string",
        "enum": [
          "html",
          "png"
        ]
      },
      "parse": {
        "type": "boolean"
      },
      "device_type": {
        "type": "string"
      },
      "markdown": {
        "type": "boolean"
      },
      "xhr": {
        "type": "boolean"
      },
      "delivery_type": {
        "type": "string"
      },
      "target_store_id": {
        "type": "string"
      },
      "delivery_zip": {
        "type": "string"
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["target_search" /* TargetSearch */]: {
    "type": "object",
    "properties": {
      "target": {
        "const": "target_search"
      },
      "query": {
        "type": "string",
        "maxLength": 2048
      },
      "headless": {
        "type": "string",
        "enum": [
          "html",
          "png"
        ]
      },
      "parse": {
        "type": "boolean"
      },
      "device_type": {
        "type": "string"
      },
      "delivery_type": {
        "type": "string"
      },
      "target_store_id": {
        "type": "string"
      },
      "delivery_zip": {
        "type": "string"
      },
      "xhr": {
        "type": "boolean"
      },
      "markdown": {
        "type": "boolean"
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["target" /* Target */]: {
    "type": "object",
    "properties": {
      "target": {
        "const": "target"
      },
      "url": {
        "type": "string",
        "maxLength": 2056
      },
      "headless": {
        "type": "string",
        "enum": [
          "html",
          "png"
        ]
      },
      "device_type": {
        "type": "string"
      },
      "xhr": {
        "type": "boolean"
      },
      "delivery_zip": {
        "type": "string"
      },
      "target_store_id": {
        "type": "string"
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["lowes_search" /* LowesSearch */]: {
    "type": "object",
    "properties": {
      "target": {
        "const": "lowes_search"
      },
      "query": {
        "type": "string",
        "maxLength": 2048
      },
      "lowes_store_id": {
        "type": "string"
      },
      "headless": {
        "type": "string",
        "enum": [
          "html",
          "png"
        ]
      },
      "delivery_zip": {
        "type": "string"
      },
      "user_agent_type": {
        "type": "string"
      },
      "free_delivery": {
        "type": "boolean"
      },
      "pickup_today": {
        "type": "boolean"
      },
      "delivery_today_tomorrow": {
        "type": "boolean"
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["universal" /* Universal */]: {
    "type": "object",
    "properties": {
      "target": {
        "const": "universal"
      },
      "url": {
        "type": "string",
        "maxLength": 2056
      },
      "payload": {
        "type": "string"
      },
      "proxy_pool": {
        "type": "string",
        "enum": [
          "standard",
          "premium"
        ]
      },
      "http_method": {
        "type": "string"
      },
      "headless": {
        "type": "string",
        "enum": [
          "html",
          "png"
        ]
      },
      "geo": {
        "type": "string"
      },
      "locale": {
        "type": "string"
      },
      "device_type": {
        "type": "string"
      },
      "session_id": {
        "type": "string"
      },
      "successful_status_codes": {
        "type": "array",
        "items": {
          "type": "number"
        }
      },
      "headers": {
        "type": "object"
      },
      "cookies": {
        "type": "object"
      },
      "force_headers": {
        "type": "boolean"
      },
      "force_cookies": {
        "type": "boolean"
      },
      "xhr": {
        "type": "boolean"
      },
      "markdown": {
        "type": "boolean"
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["chatgpt" /* Chatgpt */]: {
    "type": "object",
    "properties": {
      "target": {
        "const": "chatgpt"
      },
      "prompt": {
        "type": "string",
        "maxLength": 8192
      },
      "search": {
        "type": "boolean"
      },
      "parse": {
        "type": "boolean"
      },
      "geo": {
        "type": "string"
      },
      "device_type": {
        "type": "string"
      },
      "markdown": {
        "type": "boolean"
      },
      "xhr": {
        "type": "boolean"
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["perplexity" /* Perplexity */]: {
    "type": "object",
    "properties": {
      "target": {
        "const": "perplexity"
      },
      "prompt": {
        "type": "string",
        "maxLength": 8192
      },
      "parse": {
        "type": "boolean"
      },
      "geo": {
        "type": "string"
      },
      "device_type": {
        "type": "string"
      },
      "markdown": {
        "type": "boolean"
      },
      "xhr": {
        "type": "boolean"
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["gemini" /* Gemini */]: {
    "type": "object",
    "properties": {
      "target": {
        "const": "gemini"
      },
      "prompt": {
        "type": "string",
        "maxLength": 8192
      },
      "parse": {
        "type": "boolean"
      },
      "geo": {
        "type": "string"
      },
      "xhr": {
        "type": "boolean"
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["bbb" /* Bbb */]: {
    "type": "object",
    "properties": {
      "target": {
        "const": "bbb"
      },
      "url": {
        "type": "string",
        "maxLength": 2056
      },
      "headless": {
        "type": "string",
        "enum": [
          "html",
          "png"
        ]
      },
      "geo": {
        "type": "string"
      },
      "device_type": {
        "type": "string"
      },
      "markdown": {
        "type": "boolean"
      },
      "xhr": {
        "type": "boolean"
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["autotrader" /* Autotrader */]: {
    "type": "object",
    "properties": {
      "target": {
        "const": "autotrader"
      },
      "url": {
        "type": "string",
        "maxLength": 2056
      },
      "headless": {
        "type": "string",
        "enum": [
          "html",
          "png"
        ]
      },
      "geo": {
        "type": "string"
      },
      "device_type": {
        "type": "string"
      },
      "markdown": {
        "type": "boolean"
      },
      "xhr": {
        "type": "boolean"
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["mobile" /* Mobile */]: {
    "type": "object",
    "properties": {
      "target": {
        "const": "mobile"
      },
      "url": {
        "type": "string",
        "maxLength": 2056
      },
      "headless": {
        "type": "string",
        "enum": [
          "html",
          "png"
        ]
      },
      "geo": {
        "type": "string"
      },
      "device_type": {
        "type": "string"
      },
      "markdown": {
        "type": "boolean"
      },
      "xhr": {
        "type": "boolean"
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["airbnb" /* Airbnb */]: {
    "type": "object",
    "properties": {
      "target": {
        "const": "airbnb"
      },
      "url": {
        "type": "string",
        "maxLength": 2056
      },
      "headless": {
        "type": "string",
        "enum": [
          "html",
          "png"
        ]
      },
      "geo": {
        "type": "string"
      },
      "device_type": {
        "type": "string"
      },
      "markdown": {
        "type": "boolean"
      },
      "xhr": {
        "type": "boolean"
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["apple_app_store" /* AppleAppStore */]: {
    "type": "object",
    "properties": {
      "target": {
        "const": "apple_app_store"
      },
      "url": {
        "type": "string",
        "maxLength": 2056
      },
      "headless": {
        "type": "string",
        "enum": [
          "html",
          "png"
        ]
      },
      "geo": {
        "type": "string"
      },
      "device_type": {
        "type": "string"
      },
      "markdown": {
        "type": "boolean"
      },
      "xhr": {
        "type": "boolean"
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["instagram_graphql_profile" /* InstagramGraphqlProfile */]: {
    "type": "object",
    "properties": {
      "target": {
        "const": "instagram_graphql_profile"
      },
      "query": {
        "type": "string",
        "maxLength": 2048
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["tiktok_post" /* TiktokPost */]: {
    "type": "object",
    "properties": {
      "target": {
        "const": "tiktok_post"
      },
      "url": {
        "type": "string",
        "maxLength": 2056
      },
      "xhr": {
        "type": "boolean"
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["tiktok_shop_search" /* TiktokShopSearch */]: {
    "type": "object",
    "properties": {
      "target": {
        "const": "tiktok_shop_search"
      },
      "query": {
        "type": "string",
        "maxLength": 2048
      },
      "headless": {
        "type": "string",
        "enum": [
          "html",
          "png"
        ]
      },
      "device_type": {
        "type": "string"
      },
      "markdown": {
        "type": "boolean"
      },
      "country": {
        "type": "string"
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["tiktok_shop_product" /* TiktokShopProduct */]: {
    "type": "object",
    "properties": {
      "target": {
        "const": "tiktok_shop_product"
      },
      "product_id": {
        "type": "string"
      },
      "headless": {
        "type": "string",
        "enum": [
          "html",
          "png"
        ]
      },
      "device_type": {
        "type": "string"
      },
      "xhr": {
        "type": "boolean"
      },
      "markdown": {
        "type": "boolean"
      },
      "country": {
        "type": "string"
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["tiktok" /* Tiktok */]: {
    "type": "object",
    "properties": {
      "target": {
        "const": "tiktok"
      },
      "url": {
        "type": "string",
        "maxLength": 2056
      },
      "headless": {
        "type": "string",
        "enum": [
          "html",
          "png"
        ]
      },
      "user_agent_type": {
        "type": "string"
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["reddit_post" /* RedditPost */]: {
    "type": "object",
    "properties": {
      "target": {
        "const": "reddit_post"
      },
      "url": {
        "type": "string",
        "maxLength": 2056
      },
      "locale": {
        "type": "string"
      },
      "geo": {
        "type": "string"
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["reddit_subreddit" /* RedditSubreddit */]: {
    "type": "object",
    "properties": {
      "target": {
        "const": "reddit_subreddit"
      },
      "url": {
        "type": "string",
        "maxLength": 2056
      },
      "locale": {
        "type": "string"
      },
      "geo": {
        "type": "string"
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["reddit_user" /* RedditUser */]: {
    "type": "object",
    "properties": {
      "target": {
        "const": "reddit_user"
      },
      "url": {
        "type": "string",
        "maxLength": 2056
      },
      "locale": {
        "type": "string"
      },
      "geo": {
        "type": "string"
      },
      "sort": {
        "type": "string"
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["youtube_video" /* YoutubeVideo */]: {
    "type": "object",
    "properties": {
      "target": {
        "const": "youtube_video"
      },
      "query": {
        "type": "string",
        "maxLength": 2048
      },
      "geo": {
        "type": "string"
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["youtube_metadata" /* YoutubeMetadata */]: {
    "type": "object",
    "properties": {
      "target": {
        "const": "youtube_metadata"
      },
      "query": {
        "type": "string",
        "maxLength": 2048
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["youtube_search" /* YoutubeSearch */]: {
    "type": "object",
    "properties": {
      "360": {
        "type": "boolean"
      },
      "target": {
        "const": "youtube_search"
      },
      "query": {
        "type": "string",
        "maxLength": 2048
      },
      "upload_date": {
        "type": "string"
      },
      "type": {
        "type": "string"
      },
      "duration": {
        "type": "string"
      },
      "video_sort_by": {
        "type": "string"
      },
      "3d": {
        "type": "boolean"
      },
      "4k": {
        "type": "boolean"
      },
      "creative_commons": {
        "type": "boolean"
      },
      "hd": {
        "type": "boolean"
      },
      "hdr": {
        "type": "boolean"
      },
      "vr180": {
        "type": "boolean"
      },
      "live": {
        "type": "boolean"
      },
      "location": {
        "type": "boolean"
      },
      "purchased": {
        "type": "boolean"
      },
      "subtitles": {
        "type": "boolean"
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["youtube_search_max" /* YoutubeSearchMax */]: {
    "type": "object",
    "properties": {
      "360": {
        "type": "boolean"
      },
      "target": {
        "const": "youtube_search_max"
      },
      "query": {
        "type": "string",
        "maxLength": 2048
      },
      "upload_date": {
        "type": "string"
      },
      "type": {
        "type": "string"
      },
      "duration": {
        "type": "string"
      },
      "video_sort_by": {
        "type": "string"
      },
      "3d": {
        "type": "boolean"
      },
      "4k": {
        "type": "boolean"
      },
      "creative_commons": {
        "type": "boolean"
      },
      "hd": {
        "type": "boolean"
      },
      "hdr": {
        "type": "boolean"
      },
      "vr180": {
        "type": "boolean"
      },
      "live": {
        "type": "boolean"
      },
      "location": {
        "type": "boolean"
      },
      "purchased": {
        "type": "boolean"
      },
      "subtitles": {
        "type": "boolean"
      },
      "markdown": {
        "type": "boolean"
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["youtube_subtitles" /* YoutubeSubtitles */]: {
    "type": "object",
    "properties": {
      "target": {
        "const": "youtube_subtitles"
      },
      "query": {
        "type": "string",
        "maxLength": 2048
      },
      "language_code": {
        "type": "string"
      },
      "subtitle_origin": {
        "type": "string"
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  },
  ["youtube_channel" /* YoutubeChannel */]: {
    "type": "object",
    "properties": {
      "target": {
        "const": "youtube_channel"
      },
      "query": {
        "type": "string",
        "maxLength": 2048
      },
      "parse": {
        "type": "boolean"
      },
      "limit": {
        "type": "number",
        "minimum": 0,
        "maximum": 100
      },
      "markdown": {
        "type": "boolean"
      },
      "callback_url": {
        "type": "string"
      }
    },
    "required": [
      "target"
    ],
    "additionalProperties": false
  }
};
var requestSchemas = {
  ["universal_ecommerce" /* UniversalEcommerce */]: z.fromJSONSchema(requestJsonSchemas["universal_ecommerce" /* UniversalEcommerce */]),
  ["google_search" /* GoogleSearch */]: z.fromJSONSchema(requestJsonSchemas["google_search" /* GoogleSearch */]),
  ["google_travel_hotels" /* GoogleTravelHotels */]: z.fromJSONSchema(requestJsonSchemas["google_travel_hotels" /* GoogleTravelHotels */]),
  ["google_trends_explore" /* GoogleTrendsExplore */]: z.fromJSONSchema(requestJsonSchemas["google_trends_explore" /* GoogleTrendsExplore */]),
  ["google_shopping_search" /* GoogleShoppingSearch */]: z.fromJSONSchema(requestJsonSchemas["google_shopping_search" /* GoogleShoppingSearch */]),
  ["google_shopping_product" /* GoogleShoppingProduct */]: z.fromJSONSchema(requestJsonSchemas["google_shopping_product" /* GoogleShoppingProduct */]),
  ["google" /* Google */]: z.fromJSONSchema(requestJsonSchemas["google" /* Google */]),
  ["google_suggest" /* GoogleSuggest */]: z.fromJSONSchema(requestJsonSchemas["google_suggest" /* GoogleSuggest */]),
  ["google_maps" /* GoogleMaps */]: z.fromJSONSchema(requestJsonSchemas["google_maps" /* GoogleMaps */]),
  ["google_ai_mode" /* GoogleAiMode */]: z.fromJSONSchema(requestJsonSchemas["google_ai_mode" /* GoogleAiMode */]),
  ["google_ads" /* GoogleAds */]: z.fromJSONSchema(requestJsonSchemas["google_ads" /* GoogleAds */]),
  ["google_lens" /* GoogleLens */]: z.fromJSONSchema(requestJsonSchemas["google_lens" /* GoogleLens */]),
  ["bing_search" /* BingSearch */]: z.fromJSONSchema(requestJsonSchemas["bing_search" /* BingSearch */]),
  ["bing" /* Bing */]: z.fromJSONSchema(requestJsonSchemas["bing" /* Bing */]),
  ["youtube_transcript" /* YoutubeTranscript */]: z.fromJSONSchema(requestJsonSchemas["youtube_transcript" /* YoutubeTranscript */]),
  ["amazon_product" /* AmazonProduct */]: z.fromJSONSchema(requestJsonSchemas["amazon_product" /* AmazonProduct */]),
  ["amazon_pricing" /* AmazonPricing */]: z.fromJSONSchema(requestJsonSchemas["amazon_pricing" /* AmazonPricing */]),
  ["amazon_search" /* AmazonSearch */]: z.fromJSONSchema(requestJsonSchemas["amazon_search" /* AmazonSearch */]),
  ["amazon_sellers" /* AmazonSellers */]: z.fromJSONSchema(requestJsonSchemas["amazon_sellers" /* AmazonSellers */]),
  ["amazon_bestsellers" /* AmazonBestsellers */]: z.fromJSONSchema(requestJsonSchemas["amazon_bestsellers" /* AmazonBestsellers */]),
  ["amazon" /* Amazon */]: z.fromJSONSchema(requestJsonSchemas["amazon" /* Amazon */]),
  ["ecommerce" /* Ecommerce */]: z.fromJSONSchema(requestJsonSchemas["ecommerce" /* Ecommerce */]),
  ["walmart_product" /* WalmartProduct */]: z.fromJSONSchema(requestJsonSchemas["walmart_product" /* WalmartProduct */]),
  ["walmart_search" /* WalmartSearch */]: z.fromJSONSchema(requestJsonSchemas["walmart_search" /* WalmartSearch */]),
  ["walmart" /* Walmart */]: z.fromJSONSchema(requestJsonSchemas["walmart" /* Walmart */]),
  ["target_product" /* TargetProduct */]: z.fromJSONSchema(requestJsonSchemas["target_product" /* TargetProduct */]),
  ["target_search" /* TargetSearch */]: z.fromJSONSchema(requestJsonSchemas["target_search" /* TargetSearch */]),
  ["target" /* Target */]: z.fromJSONSchema(requestJsonSchemas["target" /* Target */]),
  ["lowes_search" /* LowesSearch */]: z.fromJSONSchema(requestJsonSchemas["lowes_search" /* LowesSearch */]),
  ["universal" /* Universal */]: z.fromJSONSchema(requestJsonSchemas["universal" /* Universal */]),
  ["chatgpt" /* Chatgpt */]: z.fromJSONSchema(requestJsonSchemas["chatgpt" /* Chatgpt */]),
  ["perplexity" /* Perplexity */]: z.fromJSONSchema(requestJsonSchemas["perplexity" /* Perplexity */]),
  ["gemini" /* Gemini */]: z.fromJSONSchema(requestJsonSchemas["gemini" /* Gemini */]),
  ["bbb" /* Bbb */]: z.fromJSONSchema(requestJsonSchemas["bbb" /* Bbb */]),
  ["autotrader" /* Autotrader */]: z.fromJSONSchema(requestJsonSchemas["autotrader" /* Autotrader */]),
  ["mobile" /* Mobile */]: z.fromJSONSchema(requestJsonSchemas["mobile" /* Mobile */]),
  ["airbnb" /* Airbnb */]: z.fromJSONSchema(requestJsonSchemas["airbnb" /* Airbnb */]),
  ["apple_app_store" /* AppleAppStore */]: z.fromJSONSchema(requestJsonSchemas["apple_app_store" /* AppleAppStore */]),
  ["instagram_graphql_profile" /* InstagramGraphqlProfile */]: z.fromJSONSchema(requestJsonSchemas["instagram_graphql_profile" /* InstagramGraphqlProfile */]),
  ["tiktok_post" /* TiktokPost */]: z.fromJSONSchema(requestJsonSchemas["tiktok_post" /* TiktokPost */]),
  ["tiktok_shop_search" /* TiktokShopSearch */]: z.fromJSONSchema(requestJsonSchemas["tiktok_shop_search" /* TiktokShopSearch */]),
  ["tiktok_shop_product" /* TiktokShopProduct */]: z.fromJSONSchema(requestJsonSchemas["tiktok_shop_product" /* TiktokShopProduct */]),
  ["tiktok" /* Tiktok */]: z.fromJSONSchema(requestJsonSchemas["tiktok" /* Tiktok */]),
  ["reddit_post" /* RedditPost */]: z.fromJSONSchema(requestJsonSchemas["reddit_post" /* RedditPost */]),
  ["reddit_subreddit" /* RedditSubreddit */]: z.fromJSONSchema(requestJsonSchemas["reddit_subreddit" /* RedditSubreddit */]),
  ["reddit_user" /* RedditUser */]: z.fromJSONSchema(requestJsonSchemas["reddit_user" /* RedditUser */]),
  ["youtube_video" /* YoutubeVideo */]: z.fromJSONSchema(requestJsonSchemas["youtube_video" /* YoutubeVideo */]),
  ["youtube_metadata" /* YoutubeMetadata */]: z.fromJSONSchema(requestJsonSchemas["youtube_metadata" /* YoutubeMetadata */]),
  ["youtube_search" /* YoutubeSearch */]: z.fromJSONSchema(requestJsonSchemas["youtube_search" /* YoutubeSearch */]),
  ["youtube_search_max" /* YoutubeSearchMax */]: z.fromJSONSchema(requestJsonSchemas["youtube_search_max" /* YoutubeSearchMax */]),
  ["youtube_subtitles" /* YoutubeSubtitles */]: z.fromJSONSchema(requestJsonSchemas["youtube_subtitles" /* YoutubeSubtitles */]),
  ["youtube_channel" /* YoutubeChannel */]: z.fromJSONSchema(requestJsonSchemas["youtube_channel" /* YoutubeChannel */])
};

// src/schema/bundled-schema.ts
var BundledSchema = class _BundledSchema {
  static shared = new _BundledSchema();
  getRequestSchema(target) {
    return requestSchemas[target];
  }
  listTargets() {
    return [...targets];
  }
  getTargetMeta(target) {
    return targetMeta[target];
  }
  getTargetParameterSchema(target) {
    return requestJsonSchemas[target];
  }
  getSharedParameters() {
    return {};
  }
};

// src/api/web-scraping-api.ts
var WebScrapingApi = class {
  http;
  schemas;
  constructor(http, schemas = BundledSchema.shared) {
    this.http = http;
    this.schemas = schemas;
  }
  validate(params) {
    const schema = this.schemas.getRequestSchema(params.target);
    if (!schema) {
      return;
    }
    const parsed = schema.safeParse(params);
    if (!parsed.success) {
      throw new ValidationError((0, import_zod.prettifyError)(parsed.error));
    }
  }
  async scrape(params) {
    this.validate(params);
    return this.http.post("/v2/scrape", params);
  }
  async scrapeAsync(params) {
    this.validate(params);
    return this.http.post("/v3/task", params);
  }
  async scrapeBatch(params) {
    return this.http.post("/v3/task/batch", params);
  }
  async getStatus(taskId) {
    return this.http.get(`/v3/task/${taskId}`);
  }
  async getResults(taskId) {
    const res = await this.http.get(
      `/v3/task/${taskId}/results`
    );
    return res ?? null;
  }
};

// src/client.ts
var WEB_API_BASE_URL = "https://scraper-api.decodo.com";
var DEFAULT_TIMEOUT_MS = 18e4;
var notConfigured = (namespace, hint) => {
  throw new Error(`${namespace} is not configured. ${hint}`);
};
var DecodoClient = class {
  webScrapingApi;
  constructor(config) {
    const timeoutMs = config.timeoutMs ?? DEFAULT_TIMEOUT_MS;
    const schema = config.schema ?? BundledSchema.shared;
    if (config.webScrapingApi) {
      this.webScrapingApi = new WebScrapingApi(
        new HttpClient({
          baseUrl: WEB_API_BASE_URL,
          auth: {
            type: "basic",
            token: config.webScrapingApi.token
          },
          timeoutMs,
          integrationHeader: config.webScrapingApi.integrationHeader
        }),
        schema
      );
    } else {
      this.webScrapingApi = new Proxy({}, {
        get() {
          notConfigured(
            "webScrapingApi",
            "Provide webScrapingApi.token in DecodoConfig."
          );
        }
      });
    }
  }
};

// src/schema/remote-schema.ts
var import_node_fs = require("node:fs");
var import_node_path2 = require("node:path");
var z2 = __toESM(require("zod"), 1);

// src/schema/build-target-meta.ts
var getTargetParameterKeys = (parameterSchema) => {
  const properties = parameterSchema.properties ?? {};
  return Object.keys(properties).filter((key) => key !== "target");
};
var buildTargetMeta = (targets2) => {
  const meta = /* @__PURE__ */ new Map();
  for (const [targetKey, target] of Object.entries(targets2)) {
    meta.set(targetKey, {
      group: target.group,
      response_format: target.response_format,
      parameters: getTargetParameterKeys(target.parameter_schema)
    });
  }
  return meta;
};

// src/schema/constants.ts
var DEFAULT_IR_PREFIX = "decodo-ir-v";
var DEFAULT_IR_BASE = "https://storage.googleapis.com/decodo-sdk-config";
var DEFAULT_IR_LIST_URL = "https://storage.googleapis.com/storage/v1/b/decodo-sdk-config/o?prefix=decodo-ir-v";
var DEFAULT_IR_CACHE_PATH = "~/.decodo/decodo.ir.json";

// src/schema/expand-path.ts
var import_node_os = require("node:os");
var import_node_path = require("node:path");
var expandPath = (path) => {
  if (path.startsWith("~/")) {
    return (0, import_node_path.resolve)((0, import_node_os.homedir)(), path.slice(2));
  }
  if (path === "~") {
    return (0, import_node_os.homedir)();
  }
  return (0, import_node_path.resolve)(path);
};

// src/schema/resolve-latest-ir.ts
var IR_OBJECT_PATTERN = /^decodo-ir-v(.+)\.json$/;
var parseSemver = (version) => {
  const match = /^(\d+)\.(\d+)\.(\d+)$/.exec(version);
  if (!match) {
    return null;
  }
  return [Number(match[1]), Number(match[2]), Number(match[3])];
};
var compareSemver = (left, right) => {
  const parsedLeft = parseSemver(left);
  const parsedRight = parseSemver(right);
  if (!parsedLeft || !parsedRight) {
    return left.localeCompare(right);
  }
  for (let i = 0; i < 3; i += 1) {
    if (parsedLeft[i] !== parsedRight[i]) {
      return parsedLeft[i] - parsedRight[i];
    }
  }
  return 0;
};
var parseIrVersions = (names) => {
  const versions = [];
  for (const name of names) {
    const match = IR_OBJECT_PATTERN.exec(name);
    if (match && parseSemver(match[1])) {
      versions.push(match[1]);
    }
  }
  return versions;
};
var resolveLatestIr = async () => {
  const res = await fetch(DEFAULT_IR_LIST_URL, {
    headers: { Accept: "application/json" }
  });
  if (!res.ok) {
    throw new Error(
      `Failed to list IR versions from ${DEFAULT_IR_LIST_URL}: HTTP ${res.status}`
    );
  }
  const body = await res.json();
  const names = body.items?.map((item) => item.name) ?? [];
  const versions = parseIrVersions(names);
  if (versions.length === 0) {
    throw new Error(
      `No versioned IR objects found in bucket with prefix "${DEFAULT_IR_PREFIX}".`
    );
  }
  const version = versions.reduce(
    (latest, current) => compareSemver(current, latest) > 0 ? current : latest
  );
  return {
    version,
    url: `${DEFAULT_IR_BASE}/${DEFAULT_IR_PREFIX}${version}.json`
  };
};

// src/schema/remote-schema.ts
var readCachedIr = (cachePath) => {
  try {
    const raw = (0, import_node_fs.readFileSync)(cachePath, "utf-8");
    return JSON.parse(raw);
  } catch {
    return null;
  }
};
var isCacheFresh = (cached, ttlMs) => {
  if (ttlMs === void 0) {
    return true;
  }
  return Date.now() - cached.fetchedAt < ttlMs;
};
var fetchIr = async (url) => {
  const res = await fetch(url, {
    headers: { Accept: "application/json" }
  });
  if (!res.ok) {
    throw new Error(`Failed to fetch IR from ${url}: HTTP ${res.status}`);
  }
  return await res.json();
};
var writeCachedIr = (cachePath, cached) => {
  (0, import_node_fs.mkdirSync)((0, import_node_path2.dirname)(cachePath), { recursive: true });
  (0, import_node_fs.writeFileSync)(cachePath, JSON.stringify(cached, null, 2));
};
var resolveIrUrl = async (config) => {
  if (config.url) {
    return { url: config.url };
  }
  const latest = await resolveLatestIr();
  return { url: latest.url, version: latest.version };
};
var loadIr = async (config) => {
  const cachePath = expandPath(config.cachePath ?? DEFAULT_IR_CACHE_PATH);
  const cached = readCachedIr(cachePath);
  const resolved = await resolveIrUrl(config);
  if (cached?.resolvedUrl === resolved.url) {
    if (isCacheFresh(cached, config.ttlMs)) {
      return cached.ir;
    }
    if (config.url || cached.ir.version === resolved.version) {
      return cached.ir;
    }
  }
  const ir = await fetchIr(resolved.url);
  writeCachedIr(cachePath, {
    fetchedAt: Date.now(),
    resolvedUrl: resolved.url,
    ir
  });
  return ir;
};
var buildRequestSchemas = (ir) => {
  const schemas = /* @__PURE__ */ new Map();
  for (const [targetKey, target] of Object.entries(
    ir.apis.webScrapingApi.targets
  )) {
    schemas.set(
      targetKey,
      z2.fromJSONSchema(
        target.parameter_schema
      )
    );
  }
  return schemas;
};
var buildParameterSchemas = (ir) => {
  const schemas = /* @__PURE__ */ new Map();
  for (const [targetKey, target] of Object.entries(
    ir.apis.webScrapingApi.targets
  )) {
    schemas.set(targetKey, target.parameter_schema);
  }
  return schemas;
};
var RemoteSchema = class _RemoteSchema {
  version;
  schemas;
  meta;
  parameterSchemas;
  sharedParameters;
  targetKeys;
  constructor(ir, schemas) {
    this.version = ir.version;
    this.schemas = schemas;
    this.meta = buildTargetMeta(ir.apis.webScrapingApi.targets);
    this.parameterSchemas = buildParameterSchemas(ir);
    this.sharedParameters = ir.apis.webScrapingApi.parameters ?? {};
    this.targetKeys = Object.keys(ir.apis.webScrapingApi.targets);
  }
  static async load(config = {}) {
    const ir = await loadIr(config);
    const schemas = buildRequestSchemas(ir);
    return new _RemoteSchema(ir, schemas);
  }
  getRequestSchema(target) {
    return this.schemas.get(target);
  }
  listTargets() {
    return this.targetKeys;
  }
  getTargetMeta(target) {
    return this.meta.get(target);
  }
  getTargetParameterSchema(target) {
    return this.parameterSchemas.get(target);
  }
  getSharedParameters() {
    return this.sharedParameters;
  }
};

// src/generated/parameters.ts
var parameterMeta = {
  "360": { type: "boolean" },
  callback_url: { type: "string" },
  query: { type: "string", maxLength: 2048 },
  headless: { type: "string", enum: ["html", "png"] },
  locale: { type: "string" },
  geo: { type: "string" },
  device_type: { type: "string" },
  page_from: { type: "number", minimum: 0, maximum: 100 },
  google_results_language: { type: "string" },
  google_tbm: { type: "string" },
  google_tbs: { type: "string" },
  parse: { type: "boolean" },
  google_nfpr: { type: "boolean" },
  google_safe_search: { type: "boolean" },
  session_id: { type: "string" },
  xhr: { type: "boolean" },
  markdown: { type: "boolean" },
  page_count: { type: "number", minimum: 1, maximum: 10 },
  date_range: { type: "string" },
  stars: { type: "number" },
  adults: { type: "number", minimum: 1, maximum: 3 },
  children: { type: "number", minimum: 1, maximum: 3 },
  search_type: { type: "string" },
  date_start: { type: "string" },
  date_end: { type: "string" },
  url: { type: "string", maxLength: 2056 },
  hotel_occupancy: { type: "string" },
  domain: { type: "string" },
  language_code: { type: "string" },
  transcript_origin: { type: "string" },
  autoselect_variant: { type: "boolean" },
  currency: { type: "string" },
  category: { type: "string" },
  merchant: { type: "string" },
  sort_by: { type: "string" },
  parser_type: { type: "string" },
  product_id: { type: "string" },
  fulfillment_type: { type: "string" },
  walmart_store_id: { type: "string" },
  delivery_zip: { type: "string" },
  store_id: { type: "string" },
  delivery_type: { type: "string" },
  target_store_id: { type: "string" },
  lowes_store_id: { type: "string" },
  user_agent_type: { type: "string" },
  free_delivery: { type: "boolean" },
  pickup_today: { type: "boolean" },
  delivery_today_tomorrow: { type: "boolean" },
  payload: { type: "string" },
  proxy_pool: { type: "string", enum: ["standard", "premium"] },
  http_method: { type: "string" },
  successful_status_codes: { type: "array", items: { type: "number" } },
  headers: { type: "object" },
  cookies: { type: "object" },
  force_headers: { type: "boolean" },
  force_cookies: { type: "boolean" },
  prompt: { type: "string", maxLength: 8192 },
  search: { type: "boolean" },
  country: { type: "string" },
  sort: { type: "string" },
  upload_date: { type: "string" },
  type: { type: "string" },
  duration: { type: "string" },
  video_sort_by: { type: "string" },
  "3d": { type: "boolean" },
  "4k": { type: "boolean" },
  creative_commons: { type: "boolean" },
  hd: { type: "boolean" },
  hdr: { type: "boolean" },
  vr180: { type: "boolean" },
  live: { type: "boolean" },
  location: { type: "boolean" },
  purchased: { type: "boolean" },
  subtitles: { type: "boolean" },
  subtitle_origin: { type: "string" },
  limit: { type: "number", minimum: 0, maximum: 100 }
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AuthenticationError,
  BundledSchema,
  DecodoClient,
  DecodoError,
  RateLimitError,
  RemoteSchema,
  Target,
  TimeoutError,
  ValidationError,
  WebScrapingApi,
  parameterMeta,
  targetMeta,
  targets
});
